src/centralstore/incremental/PropertyLink.cpp

```diff
@@ -0,0 +1,186 @@
+ *          // Try to insert property set to next link
+ *          this->next().insert(name,value);
+ *      } else { // If no next, That means this is leaf or at the end of the link, So add a new link and update the
+ * current link's next property address
+ *          // attach new link
+ *      }
+ *  }
+ * if propertyAddress was 0 before inserting you will get head address of the link in return value
+ *  else either updated link address or last appended link address will be returned
+ * **/
+unsigned int PropertyLink::insert(std::string name, char* value) {
+    char dataName[PropertyLink::MAX_NAME_SIZE] = {0};
+    char dataValue[PropertyLink::MAX_VALUE_SIZE] = {0};
+    std::strcpy(dataName, name.c_str());
+    std::memcpy(dataValue, value,
+                PropertyLink::MAX_VALUE_SIZE);  // strcpy or strncpy get terminated at null-character hence using memcpy
+
+    property_link_logger.debug("Received name = " + name);
+    property_link_logger.debug("Received value = " + std::string(value));
+    unsigned int nextAddress = 0;
+
+    if (this->name == name) {
+        // TODO[tmkasun]: update existing property value
+        property_link_logger.error("Property key/name already exist key = " + std::string(name));
+        return this->blockAddress;
+    } else if (this->next()) {  // Traverse to the edge/end of the link list
+        return this->next()->insert(name, value);
+    } else {  // No next link means end of the link, Now add the new link
+        property_link_logger.debug("Next prop index = " + std::to_string(PropertyLink::nextPropertyIndex));
+
+        unsigned int newAddress = PropertyLink::nextPropertyIndex * PropertyLink::PROPERTY_BLOCK_SIZE;
+        this->propertiesDB->seekp(newAddress);
+        this->propertiesDB->write(dataName, PropertyLink::MAX_NAME_SIZE);
+        this->propertiesDB->write(reinterpret_cast<char*>(dataValue), PropertyLink::MAX_VALUE_SIZE);
+        if (!this->propertiesDB->write(reinterpret_cast<char*>(&nextAddress), sizeof(nextAddress))) {
+            property_link_logger.error("Error while inserting a property " + name + " into block address " +
+                                       std::to_string(newAddress));
+            return -1;
+        }
+        this->propertiesDB->flush();
+
+        this->nextPropAddress = newAddress;
+        this->propertiesDB->seekp(this->blockAddress + PropertyLink::MAX_NAME_SIZE +
+                                  PropertyLink::MAX_VALUE_SIZE);  // seek to current property next address
+        if (!this->propertiesDB->write(reinterpret_cast<char*>(&newAddress), sizeof(newAddress))) {
+            property_link_logger.error("Error while updating  property next address for " + name +
+                                       " into block address " + std::to_string(this->blockAddress));
+            return -1;
+        }
+
+        PropertyLink::nextPropertyIndex++;  // Increment the shared property index value
+        return newAddress;
+    }
+}
+
+/**
```

Comment:
We should have more descriptive function comment here describing the parameters and the return type